src/AbstractSQLCompiler.ts

```diff
@@ -448,6 +451,14 @@ export interface Trigger {
 	level: 'ROW' | 'STATEMENT';
 	when: 'BEFORE' | 'AFTER' | 'INSTEAD OF';
 }
+export interface Index {
+	type: string;
+	fields: string[];
+	name?: string;
+	description?: string;
+	nulls?: string;
+	predicate?: BooleanTypeNodes;
```

Comment:
I got the name from the PG CREATE INDEX synopsis.
See: https://www.postgresql.org/docs/16/sql-createindex.html